frontend/model/contracts/group.js

```diff
@@ -120,6 +120,8 @@ DefineContract({
         expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
       }),
       process (state, { data, meta, hash }) {
+        // TODO/BUG: Avoid proposing same member twice.
+        // Should it be done here or on component before calling sbp?
```

Comment:
This will be a frequently asked question, because it will come up a lot.

The answer is the same as for writing bullet-proof server/client architecture: the server should always validate client input, and never expect the client to correctly do validation. It should expect clients to be malicious. For us, the "server" is the contract code here in this file. So yes, it should check here IF it is a problem that there are two invites for the same user.

I'm guessing that it is a problem, so it should include a check, and if the check fails, it should fail without throwing an exception, since this particular error isn't necessarily indicative of a major error, or an attack. In other words, it should fail the same way `'gi.contracts/group/inviteDecline'` fails:

```js
        if (invite.status !== 'valid') {
          console.error(`inviteDecline: invite for ${meta.username} is: ${invite.status}`)
          return
        }
```

*And* the client should also do a validation check (in this case, the "client" is the Vue UI code)

Reply:
makes sense! I'll do that for now on